src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -307,20 +307,26 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
         HostingEventSource.Log.RequestStart(httpContext.Request.Method, httpContext.Request.Path);
     }
 
-    [MethodImpl(MethodImplOptions.NoInlining)]
-    private Activity? StartActivity(HttpContext httpContext, bool loggingEnabled, bool diagnosticListenerActivityCreationEnabled, out bool hasDiagnosticListener)
+    private static bool TryParseActivityContext(string traceParent, string traceState, bool isRemote, out ActivityContext context)
     {
-        var activity = _activitySource.CreateActivity(ActivityName, ActivityKind.Server);
-        if (activity is null && (loggingEnabled || diagnosticListenerActivityCreationEnabled))
+        if (ActivityContext.TryParse(traceParent, traceState, out context))
```

Comment:
probably worth adding a comment that this could be simplified with the TryParse overload we added in .NET 7 as soon as you don't need to run on 6 any longer.

Reply:
Wait. This PR is for `main` right? Don't we have access to the most up-to-date .NET 7 preview versions of System.Diagnostics.DiagnosticSource?

https://github.com/dotnet/aspnetcore/blob/9a81d79124c46fa883132b48715ab3a08b7425a5/eng/Dependencies.props#L72

Can we update this to use the right `TryParse` overload and `isRemote: true` here and change only do the more complicated thing in `release/6.0`?

Reply:
Yea, I was on the fence about this at first because I wanted to minimize the delta, but now that it's done it seems simple enough that it's probably worth it.

Reply:
Done.